File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/CBPacketDisconnect.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.net.tcp.packet.PacketHandler;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;

import java.io.IOException;

public class CBPacketDisconnect extends IPacket<PacketHandler>{
    
    public static final int PACKET_ID = 3;
    
    public CBPacketDisconnect(){
        super(PACKET_ID);
    }
    
    
    public String reasonComponent;
    
    public CBPacketDisconnect(String reasonComponent){
        this();
        this.reasonComponent = reasonComponent;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeUTF(reasonComponent);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        reasonComponent = stream.readUTF();
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/CBPacketDisconnect.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.net.tcp.packet.PacketHandler
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class CBPacketDisconnect() : IPacket<PacketHandler?>(PACKET_ID) {
    var reasonComponent: String? = null

    constructor(reasonComponent: String?) : this() {
        this.reasonComponent = reasonComponent
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeUTF(reasonComponent)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        reasonComponent = stream!!.readUTF()
    }

    companion object {
        const val PACKET_ID = 3
    }
}
